B-20334 I-13046 Remove extra spaces from LOA string INT

traskowskycaci opened the pull request:
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20334)

## [Issue ticket](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A976447)

## Summary

Cleaned up the LOA string shown in the UI. Was showing blank spaces between asterisks in the staging environment which we don't want.

![image](https://github.com/transcom/mymove/assets/147537467/81e94303-6ae6-4c82-b5fd-bd63837b419e)

The cleaned up string should match an example from a previous 858:
![image](https://github.com/transcom/mymove/assets/147537467/70b813c7-13c6-423d-b977-1482f9c9e4eb)

To test:
In the staging environment, will need to test with:
issue date = 01 May 2024
Dept Ind = Navy and Marine Corps
TAC = NAD4

It should spit out the cleaned up version of the string.

robot-mymove commented:
<!--
  0 failure: 
  1 warning:  This PR does not ...
  
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR does not include changes to unit tests, even though it affects app code.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 526bbad1384c61a4a2cf0084c14facfda2457f75
</p>

deandreJones commented:
Could probably add a test or 2 for that regex check

traskowskycaci commented:
> Could probably add a test or 2 for that regex check

Added in tests, and also another regex condition for spaces before the asterisk. Broke it out into two lines for readability. @TevinAdams  re-requesting a review from you since I added something new

Can test this on the dev env with
```
{
  "serviceMemberAffiliation": "ARMY",
  "ordersIssueDate": "2014-01-01",
  "tacCode": "E12A"
}
```

Before, extra space after '1' at beginning of string:
<img width="376" alt="image" src="https://github.com/transcom/mymove/assets/147537467/f0fae76d-6017-4c21-88d7-ef8011321980">

After, no extra space after '1' at beginning of string:
<img width="404" alt="image" src="https://github.com/transcom/mymove/assets/147537467/bfe0767e-38fa-4587-9001-e8e81efcdfcb">
